tests/Mono.Android-Tests/Android.Widget/CustomWidgetTests.cs

```diff
@@ -1,20 +1,23 @@
-﻿using System.Diagnostics.CodeAnalysis;
-using Android.App;
+﻿using Android.App;
 using Android.Content;
 using Android.Util;
 using Android.Views;
 using Android.Widget;
 using NUnit.Framework;
-using Mono.Android_Test.Library;
 
 namespace Xamarin.Android.RuntimeTests
 {
 	[TestFixture]
 	public class CustomWidgetTests
 	{
+		public CustomWidgetTests()
+		{
+			// NOTE: ensure the C# compiler has a reference to the library
+			new Mono.Android_Test.Library.Foo ();
```

Comment:
Is this (still?) a problem? https://github.com/xamarin/xamarin-android/pull/8904 updated things to look for "transitive assembly references" ([see also](https://github.com/xamarin/xamarin-android/pull/4225#issuecomment-583139797)).

If we need to add explicit type references to work around missing transitive assembly references, maybe we need to revisit #8904?

Reply:
#8904 is unrelated.

The problem here is if you add a reference to a class library not using any types from it, and one of the `AndroidResource` `.xml` files uses a custom view. The C# compiler will ignore the assembly, but the `AndroidResource` files would still make it to the app.

I'm not sure if this would happen in practice/real customer projects.

Reply:
I filed this for the future, and put it in comments:

* https://github.com/xamarin/xamarin-android/issues/9008